pysat/_instrument.py

```diff
@@ -384,34 +379,69 @@ def __setitem__(self, key, new):
         long_name = 'name', and units = ''.
         
         """
+        
+        # add data to main pandas.DataFrame, depending upon the input
+        
         if isinstance(new, dict):
+            # input dict must have data in 'data', the rest
+            # is presumed to be metadata
             # metadata should be included in dict
-            self.data[key] = new.pop('data')
-            # pass the rest to meta
+            in_data = new.pop('data')
+            if isinstance(in_data, pds.DataFrame):
+                # input is a dataframe
+                pass
+ 
+            elif isinstance(in_data[0], pds.DataFrame):
+                # input is a list_like of frames
+                # this is higher order data
+                # this process ensures
+                if key not in self.meta.keys_nD():
+                    ho_meta = _meta.Meta()
+                    for ho_key in in_data[0]:
+                        ho_meta[ho_key] = {}
+                    self.meta[key] = ho_meta
+            
+            # assign data and any extra metadata
+            self.data[key] = in_data
             self.meta[key] = new
+
         else:
+            # not a dictionary, could have a mixed input
+            # aka slice, and a name
             if isinstance(key, tuple):
                 self.data.ix[key[0], key[1]] = new
                 self.meta[key[1]] = {}
             # If list or series of df handle ho data
             elif hasattr(new, '__getitem__'):
-                if isinstance(new, Series):
-                    self.data[key] = new  
-                    self.meta[key] = {}
-                elif isinstance(new, DataFrame):
+                # if isinstance(new, Series):
```

Comment:
Do you want to leave old code commented out?  I've seen it in a few other places and it does make things a bit jarring to read.  Do you think you'll be switching back at some point?

Reply:
Bad habit started when I didn't have git. When I make a bunch of changes I keep the commented code around for a bit in case I need to retain something.

I deleted those comments in a pull request to this same branch.